nodejs/lancedb/table.ts

```diff
@@ -478,4 +478,12 @@ export class Table {
   async toArrow(): Promise<ArrowTable> {
     return await this.query().toArrow();
   }
+  /** List all the stats of a specified index
+   * @param {string} indexUUID The UUID of the index to get stats for
+   * @returns {IndexStatistics | undefined} The stats of the index
+   * If the index does not exist, `undefined` is returned
+   */
+  async indexStats(indexUUID: string): Promise<IndexStatistics | undefined> {
+    return await this.inner.indexStats(indexUUID);
+  }
```

Comment:
So the `indexStats()` method in the existing SDKs was actually incorrectly designed. The UUIDs here are meant to be internal ids, and users should instead use the `name` of the index.

This is because there can be multiple "parts" of an index, and each have their own internal UUID. But the name of the index is what they all have in common.

If you look at the underlying code to get the stats, you'll notice that it actually just uses the UUID to get the name of the index, and uses that to retrieve the stats. (`dataset.index_statistics(&index.unwrap().index_name)`.)

https://github.com/lancedb/lancedb/blob/27e4ad3f11117965fd4b57ce4c84808dfd5b8335/rust/lancedb/src/table.rs#L1252-L1262

I think it's best we just use name here from the start. We can fix the other SDKs in a follow-up PR.

Reply:
@wjones127 does this mean that all of these public methods that accept `index_uuid` should instead take in `index_name` as well? 

- [count_indexed_rows](https://github.com/universalmind303/lancedb/blob/2b040f7f00b59554d0154dfba7854b3edc60225f/rust/lancedb/src/table.rs#L1209)
- [count_unindexed_rows](https://github.com/universalmind303/lancedb/blob/2b040f7f00b59554d0154dfba7854b3edc60225f/rust/lancedb/src/table.rs#L1216)
- [get_index_type](https://github.com/universalmind303/lancedb/blob/2b040f7f00b59554d0154dfba7854b3edc60225f/rust/lancedb/src/table.rs#L1230)
- [get_distance_type](https://github.com/universalmind303/lancedb/blob/2b040f7f00b59554d0154dfba7854b3edc60225f/rust/lancedb/src/table.rs#L1230)

Reply:
Yes, though should have. Also they should not have been separate methods, because they are so often called together and therefore make duplicate requests. 🤦